src/agents/query_engine/PatternMatchingQueryProcessor.cc

```diff
@@ -73,8 +73,7 @@ void PatternMatchingQueryProcessor::update_attention_broker_single_answer(
         // Updates joint answer (stimulation)
         joint_answer.insert(handle);
         // Gets targets and stack them
-        shared_ptr<atomdb_api_types::HandleList> query_result =
-            db->query_for_targets((char*) handle.c_str());
+        auto query_result = db->query_for_targets(handle);
```

Comment:
Please keep the actual query_result type here (auto -> shared_ptr<atomdb_api_types::HandleList>)